### DISCUSSION TITLE
Custom node that opens preview in new browser tab

### DISCUSSION THREAD

**Discussion by ChrisAcrobat** [Comfy-Org/ComfyUI#2804] [Q&A] [answered]
Hello. 👋
I'm trying to develop a custom node that instead of previewing the image in a node it's opens a new browser tab and post the image to it for preview.
My idea was to write it in JavaScript on the client side and use `postMessage()`, but after a quick look around it looks like there is no possibility to execute code on client side. Is this correct?
My goal was to write a gallery-like page.

**Comment by blepping** [comment, marked as answer]
bad news or maybe good news - it already exists: https://github.com/mape/ComfyUI-mape-Helpers

**Reply by ChrisAcrobat** [reply]
Awesome news! Thanks.
If it doesn't suit me then I at the least know where to find inspiration
from. 👍
